### PR TITLE
add conditional approach for namespace create

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

This change introduces more granular config for namespace pools to optionate namespace creation and management , additionally when namespace management is disabled it will switch to configmap based labelling query in houston.

## Related Issues

with existing approach  when namespace pools  create set to true namespace will be auto created with roles with subsequent deloyment approach the status indicator will be broken showing status indicating deploying....

this change will fix these behaviours
Related https://github.com/astronomer/issues/issues/6582

## Testing
QA should able to verify global.features.namespacePools.enabled: true and global.features.namespacePools.createRbac: true/false changes

## Merging

cherry-pick to release-0.36

### CODE REVIEW

**Review by Shubham-astro — APPROVED** [review]
Looks good, we already have testcases added.

**Review by rishkarajgi — APPROVED** [review]

File: tests/chart_tests/test_houston_configmap.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def common_test_cases(docs):
     prod = yaml.safe_load(doc["data"]["production.yaml"])
 
     assert prod["deployments"]["helm"]["airflow"]["useAstroSecurityManager"] is True
-    assert prod["deployments"]["disableManageClusterScopedResources"] is False
+    assert "disableManageClusterScopedResources" not in prod["deployments"]
```
Comment (danielhoherd) [inline tests/chart_tests/test_houston_configmap.py:24]: I don't think this test is good enough. The value of `disableManageClusterScopedResources` could be true or false. If we change the configuration in the future and do not update this test, this test may begin to give a false positive.

File: charts/astronomer/templates/houston/houston-configmap.yaml [hunk 2]
```diff
@@ -71,7 +71,9 @@ data:
       {{- end }}
     # Airflow deployment configuration
     deployments:
-      disableManageClusterScopedResources: {{ .Values.global.disableManageClusterScopedResources }}
+      {{ if or .Values.global.disableManageClusterScopedResources  .Values.global.features.namespacePools.enabled }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/houston/houston-configmap.yaml:74]:
```suggestion
      {{ if or .Values.global.disableManageClusterScopedResources .Values.global.features.namespacePools.enabled }}
```

File: charts/astronomer/templates/commander/commander-rolebinding.yaml [hunk 3]
```diff
@@ -5,17 +5,17 @@
 # 1. Do not create anything is rbacEnabled is disabled
 # 2. Create ClusterRoleBinding if clusterRoles enabled and namespacePools disabled
 # 3. Create RoleBinding for each namespace in the namespacePool (+ astronomer namespace) if enabled
-{{- $namespacepoolsRBACresources:= and .Values.global.features.namespacePools.createRbac .Values.global.features.namespacePools.enabled}}
-{{- $useClusterRoles := and .Values.global.rbacEnabled .Values.global.clusterRoles (not .Values.global.features.namespacePools.enabled)}}
-{{- $shouldCreateResources := and .Values.global.rbacEnabled (or .Values.global.clusterRoles $namespacepoolsRBACresources) }}
+{{- $namespacepoolsRBACresources:= and .Values.global.features.namespacePools.createRbac .Values.global.features.namespacePools.enabled }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/commander/commander-rolebinding.yaml:8]:
```suggestion
{{- $namespacepoolsRBACresources := and .Values.global.features.namespacePools.createRbac .Values.global.features.namespacePools.enabled }}
```

**Review by danielhoherd — APPROVED** [review]
Looks good. We reviewed this feature on a call today and I'm happy with it.

**Comment by rujhan-arora-astronomer** [comment]
Looks good to me!

**Pull request merged**